pkg/hubble/relay/server/server_test.go

```diff
@@ -0,0 +1,310 @@
+	)
+	require.NoError(t, err)
+
+	m := s.GetEventsChannel()
+
+	for i := 0; i < numFlows; i++ {
+		tn := monitor.TraceNotifyV0{Type: byte(monitorAPI.MessageTypeTrace)}
+		src := getRandomEndpoint()
+		dst := getRandomEndpoint()
+		srcMAC, _ := net.ParseMAC(fake.MAC())
+		dstMAC, _ := net.ParseMAC(fake.MAC())
+		data := testutils.MustCreateL3L4Payload(tn,
+			&layers.Ethernet{
+				SrcMAC:       srcMAC,
+				DstMAC:       dstMAC,
+				EthernetType: layers.EthernetTypeIPv4,
+			},
+			&layers.IPv4{
+				SrcIP:    src.IPv4,
+				DstIP:    dst.IPv4,
+				Protocol: layers.IPProtocolTCP,
+			},
+			&layers.TCP{
+				SrcPort: layers.TCPPort(fake.Port()),
+				DstPort: layers.TCPPort(fake.Port()),
+				ACK:     true,
+				PSH:     i%4 == 0,
+			})
+		event := &observerTypes.MonitorEvent{
+			Timestamp: time.Unix(int64(i+1), 0),
+			NodeName:  nodeName,
+			Payload: &observerTypes.PerfEvent{
+				Data: data,
+				CPU:  0,
+			},
+		}
+		m <- event
+	}
+
+	return s
+}
+
+func newHubblePeer(t testing.TB, ctx context.Context, address string, hubbleObserver *observer.LocalObserverServer) {
+	options := []serveroption.Option{
+		serveroption.WithInsecure(),
+		serveroption.WithUnixSocketListener(address),
+		serveroption.WithObserverService(hubbleObserver),
+	}
+
+	srv, err := server.NewServer(log, options...)
+	require.NoError(t, err)
+
+	go func() {
+		if err := srv.Serve(); err != nil {
+			t.Log(err)
+			t.Fail()
+		}
+	}()
+
+	go func() {
+		<-ctx.Done()
+		close(hubbleObserver.GetEventsChannel())
+		<-hubbleObserver.GetStopped()
+		srv.Stop()
+	}()
+}
+
+func benchmarkRelayGetFlows(b *testing.B, withFieldMask bool) {
+	tmp, err := os.MkdirTemp("", "hubble")
+	require.NoError(b, err)
+	defer os.RemoveAll(tmp)
+	root := "unix://" + filepath.Join(tmp, "peer-")
+	ctx := context.Background()
+	numFlows := b.N
+	numPeers := 2
+
+	// FIXME: number of peers should be constant so that it scales linearly with b.N
+	if numFlows > 65535*2 {
+		numPeers = numFlows/65535 + 1
+	}
+
+	// Create hubble servers listening on unix sockets in temporary directory.
+	type peer struct {
+		name     string
+		address  string
+		observer *observer.LocalObserverServer
+	}
+	peers := make([]peer, numPeers)
+	flowsScheduled := 0
+	for i := range peers {
+		address := fmt.Sprintf("%s%d.sock", root, i)
+		name := fake.K8sNodeName()
+		numFlowsPerPeer := numFlows / len(peers)
+		if i == len(peers)-1 {
+			numFlowsPerPeer = numFlows - flowsScheduled
+		}
+		// can't retrieve one last flow from the buffer
+		hubbleObserver := newHubbleObserver(b, name, numFlowsPerPeer+1)
+		newHubblePeer(b, ctx, address, hubbleObserver)
+		flowsScheduled += numFlowsPerPeer
+		peers[i] = peer{name, address, hubbleObserver}
+		go hubbleObserver.Start()
+	}
+
+	// Create hubble relay server and connect to all peers from previous step.
+	ccb := pool.GRPCClientConnBuilder{
+		DialTimeout: defaults.DialTimeout,
+		Options: []grpc.DialOption{
+			grpc.WithInsecure(),
```

Comment:
`WithInsecure()` is deprecated and can be removed anytime. Let's not use it.
```suggestion
			grpc.WithTransportCredentials(insecure.NewCredentials()),
```